fms_core/management/commands/curation.py

```diff
@@ -0,0 +1,93 @@
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction, IntegrityError
+import json
+from os.path import expanduser
+import os
+import time
+import datetime
+import logging
+import reversion
+from shutil import copyfile
+from django.contrib.auth.models import User
+# Import the functions of the various curations available.
+from ._rollback_extraction import rollback_extraction
+from ._rollback_curation import rollback_curation
+
+# Available actions
+ACTION_ROLLBACK_CURATION = "rollback_curation"
+ACTION_ROLLBACK_EXTRACTION = "rollback_extraction"
+
+# Curation params template
+# [CURATION_ACTION_TEMPLATE_1,CURATION_ACTION_TEMPLATE_2,...]
+
+# Other contants
+HOME = expanduser("~")
+CURATION_PATH = "/curation/"
+LOG_PATH = "log/"
+
+
+class Command(BaseCommand):
+    help = 'Manage curations'
+
+    curation_switch = {
+        ACTION_ROLLBACK_EXTRACTION: rollback_extraction,
+        ACTION_ROLLBACK_CURATION: rollback_curation
+    }
+
+    def init_logging(self, log_name, timestamp):
+        filename = HOME + CURATION_PATH + LOG_PATH + timestamp + "_" + log_name + ".log"
+        formatter = logging.Formatter("%(asctime)s || %(levelname)s || %(message)s", datefmt="%Y-%m-%d %H:%M:%S")
+        handler = logging.FileHandler(filename, "w+")
+        handler.setFormatter(formatter)
+        log = logging.getLogger(log_name)
+        log.setLevel(logging.DEBUG)
+        log.addHandler(handler)
+        return log
+
+    def add_arguments(self, parser):
+        parser.add_argument("-p", dest="params_file", help="curation parameters file (json format)", required=True)
+
+    def handle(self, *args, **options):
+        self.stdout.write(self.style.SUCCESS("Started curation."))
+        # Copy the curation file to log directory with current timestamp matching log file
+        os.environ["TZ"] = "America/Montreal"
+        time.tzset()
+        timestamp = time.strftime("%Y-%m-%d_%H:%M:%S")
```

Comment:
the timestamp is used below to construct a filename but since it uses the colon it throws the error for windows users, maybe there is some workaround I am not aware of (this would fit for all `strftime('%Y-%m-%d-%H-%M-%S')`)